### PR TITLE
WIP adds mutations for removing songs from the users library

### PR DESCRIPTION
**Pull request by DLavin23**
@sisk @tshuck

### CODE REVIEW

File: app/graphql/mutations/delete_song_user.rb [hunk 1]
```diff
@@ -0,0 +1,16 @@
+module Mutations
+  class DeleteSongUser < Mutations::BaseMutation
+    argument :song_id, ID, required: true
+
+    field :errors, [String], null: true
+
+    def resolve(song_id:)
+      song_user = SongUser.find_by(song_id: song_id, user: context[:current_user])
+      return {errors: ["Can't find song to delete"]} if song_user.blank?
```
Comment (sisk) [inline app/graphql/mutations/delete_song_user.rb:9]: Does this return a 404 response code?

**Review by sisk — COMMENTED** [review]
Main question: does the mutation for deleting need to return something other than an empty hash with an empty errors set? I would assume the API to work like:

something not found: 404
something blew up: 500
delete happened: 204

**Comment by sisk** [comment]
but...my GraphQL-fu not so good.

**Comment by tshuck** [comment]
@sisk 

It's a good question -- GraphQL seems to want to always return OK and populate the `errors` field as necessary.  I'm not really sold on that approach, but we haven't really done any error handling yet so.... decision deferred.

**Pull request merged**